src/app/pages/details-texts/details-texts.component.ts

```diff
@@ -98,20 +102,36 @@ export class DetailsTextsComponent implements OnInit {
     private getDataService: GetDataService,
     private breadcrumbsService: HandleBreadcrumbsService,
     private sanitizer: DomSanitizer,
-    private router: Router
+    private router: Router,
+    private route: ActivatedRoute
   ) {
     this.breadcrumbsService.setBreadcrumbs(this.breadcrumbLink);
   }
 
   ngOnInit() {
-    this.getDataService.getTermData().subscribe((data) => {
-      this.handleTextToHTMLConversion(data, true);
+    this.route.paramMap.subscribe((paramMap: ParamMap) => {
+      this.paramMap = paramMap;
```

Comment:
Out of curiosity, why do we need to subscribe for this instead of assigning it directly? Would the route's paramMap not be available from the start? (wondering if this might avoid a race condition with the `get` just below)

Reply:
Good point, I've reverted back to the synchronous version